tests/unit/rpc/RPCHelpersTests.cpp

```diff
@@ -547,6 +556,264 @@ TEST_F(RPCHelpersTest, ParseIssue)
+{
+    auto account = getAccountIdWithString(kACCOUNT);
+    auto account2 = getAccountIdWithString(kACCOUNT2);
+
+    auto const usdRippleState =
+        createRippleStateLedgerObject("USD", kACCOUNT2, 100, kACCOUNT, 100, kACCOUNT2, 100, kTXN_ID, 3);
+    auto const usdRippleStateKk = ripple::keylet::line(account2, account, ripple::to_currency("USD")).key;
+
+    // trustline fetched twice. once in accountHolds and once in isFrozen
+    EXPECT_CALL(*backend_, doFetchLedgerObject(usdRippleStateKk, testing::_, testing::_)).Times(2);
+    ON_CALL(*backend_, doFetchLedgerObject(usdRippleStateKk, testing::_, testing::_))
+        .WillByDefault(testing::Return(usdRippleState.getSerializer().peekData()));
+
+    EXPECT_CALL(*mockAmendmentCenterPtr_, isEnabled(testing::_, Amendments::fixFrozenLPTokenTransfer, testing::_))
+        .Times(1);
+    ON_CALL(*mockAmendmentCenterPtr_, isEnabled(testing::_, Amendments::fixFrozenLPTokenTransfer, testing::_))
+        .WillByDefault(testing::Return(true));
+
+    auto const account2Kk = ripple::keylet::account(account2).key;
+    auto const account2Root = createAccountRootObject(kACCOUNT2, 0, 2, 200, 2, kINDEX1, 2, 0);
+
+    EXPECT_CALL(*backend_, doFetchLedgerObject(account2Kk, testing::_, testing::_)).Times(2);
+    ON_CALL(*backend_, doFetchLedgerObject(account2Kk, testing::_, testing::_))
+        .WillByDefault(Return(account2Root.getSerializer().peekData()));
+
+    boost::asio::spawn(ctx_, [&, this](boost::asio::yield_context yield) {
+        auto ret = accountHolds(
+            *backend_, *mockAmendmentCenterPtr_, 0, account, ripple::to_currency("USD"), account2, true, yield
+        );
+        EXPECT_EQ(ret.mantissa(), 1000000000000000);
+    });
+    ctx_.run();
+}
+
+TEST_F(RPCHelpersTest, AccountHoldsLPTokenAsset1Frozen)
+{
+    auto ammAccount = getAccountIdWithString(kAMM_ACCOUNT);
+    auto account = getAccountIdWithString(kACCOUNT);
+    auto issuer = getAccountIdWithString(kISSUER);
+
+    auto const lptRippleState = createRippleStateLedgerObject(
+        kLPTOKEN_CURRENCY, kAMM_ACCOUNT, 100, kACCOUNT, 100, kAMM_ACCOUNT, 100, kTXN_ID, 3
+    );
+    auto const lptRippleStateKk = ripple::keylet::line(ammAccount, account, ripple::to_currency(kLPTOKEN_CURRENCY)).key;
+
+    // trustline fetched twice. once in accountHolds and once in isFrozen
+    EXPECT_CALL(*backend_, doFetchLedgerObject(lptRippleStateKk, testing::_, testing::_)).Times(2);
+    ON_CALL(*backend_, doFetchLedgerObject(lptRippleStateKk, testing::_, testing::_))
+        .WillByDefault(testing::Return(lptRippleState.getSerializer().peekData()));
+
+    EXPECT_CALL(*mockAmendmentCenterPtr_, isEnabled(testing::_, Amendments::fixFrozenLPTokenTransfer, testing::_))
+        .Times(1);
+    ON_CALL(*mockAmendmentCenterPtr_, isEnabled(testing::_, Amendments::fixFrozenLPTokenTransfer, testing::_))
+        .WillByDefault(testing::Return(true));
+
+    auto const ammID = ripple::uint256{54321};
```

Comment:
We usually add constants like these at the top of the file in an anonymous namespace and make them constexpr

Reply:
fixed